WordPress/pattern-directory
# Theme: Add copy functionality to the block preview.

This PR allows users to copy block patterns via the block preview "copy" button. 

As mentioned in the issue, there isn't a design for this so I took a crack at the experience. Since there is very little space to confirm to the user that the action has succeeded, I decided to update the copy button on success from `copy` to `copied` and set a timeout to return to the original text after 3 seconds. 

**Not so great**:
It isn't clear to me what to do when copies fail. I've done some research and the command is fairly reliable. 
It can fail if the copied string is too long but it shouldn't really be triggered by patterns. Additionally, should the copy fail, the lack of response may be enough of a cue? We could turn the button red and the copy to 'Copy failed', maybe. 


**Alternatives:**
a. Same approach with the `setTimeout` but have a more visual change, maybe using color or an icon.
b. Leverage a library to display a `Toast` that typically appears in the bottom right/left corner of the browser window saying something like "Copied pattern."




Fixes: #88 



### Screenshots

![](https://d.pr/i/nFk8F1.gif)
### How to test the changes in this Pull Request:

1. Create a pattern
2. Visit any archive page
3. Hover over a block preview and click "Copy"
4. Notice that the button text has changed to "Copied"
5. Wait 3 seconds, notice the button copy has changed back to "Copy"
6. CTRL + V somewhere and expect to see pattern code.
